.changeset/tall-beans-suffer.md

```diff
@@ -0,0 +1,7 @@
+---
+'xstate': minor
+---
+
+The machine can now be safely JSON-serialized, using `JSON.stringify(machine)`. The shape of this serialization is defined in `machine.schema.json` and reflected in `machine.definition`.
```

Comment:
I think you have meant that `machine.defiinition` can be safely serialized - as that is what you are doing in tests:
https://github.com/davidkpiano/xstate/blob/55aa589648a9afbd153e8b8e74cbf2e0ebf573fb/packages/core/test/json.test.ts#L85

I would be in favor of keeping `machine.definition` serializable, rather than the machine itself. IMHO makes it more explicit what actually gets serialized, especially that toJSON put on a Machine would just proxy to serializing `.definition`